_posts/2020-07-29-#17977.md

```diff
@@ -0,0 +1,107 @@
+of BIP 340-342.  We've previously looked at:
+
+- [O(1) OP_IF/NOTIF/ELSE/ENDIF script implementation / Abstract out script
+  execution out of VerifyWitnessProgram()](./16902.html)
+- [Initialize PrecomputedTransactionData in CheckInputScripts and Move
+  single-sig checking EvalScript code to EvalChecksig](./18401.html)
+
+Those preliminary changes have been merged into the master branch in separate PRs.
+
+This week, we'll look at one commit from PR 17977 - _[Implement Taproot
+signature
+hashing](https://github.com/bitcoin-core-review-club/bitcoin/commit/41d08f5d77f52bec0e31bb081d85fff2d67d0467)_.
+
+**Participants are encouraged to review [the notes from the session on
+PR18401](./18401.html) in particular, since we'll be covering a lot of the same
+topics this week.**
+
+## Notes
+
+- The _transaction digest_ that is used in signature creation and verification
+  is calculated from different parts of the transaction. There are extensive
+  notes covering this topic from [our session on
+PR18401](https://bitcoincore.reviews/18401.html#pr-18401-initialize-precomputedtransactiondata-in-checkinputscripts).
+
+- BIP 341 contains [the specification of the new signature validation
+  rules](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#signature-validation-rules).
+
+- The [Taproot BIP Review series](https://github.com/ajtowns/taproot-review/)
+  covered this topic in [week
+  4](https://github.com/ajtowns/taproot-review/blob/master/week-4.md#signature-details).
+  [The meeting
+  log](http://www.erisian.com.au/meetbot/taproot-bip-review/2019/taproot-bip-review.2019-11-26-19.01.log.html)
+  from that review session has more details.
+
+- This week, we'll review the implementation of the Taproot transaction hashing
+  algorithm. We won't look at full signature validation. It may be useful to
+  checkout just up to the commit _Implement Taproot signature hashing (BIP 341)_.
+  Specifically, we'll be looking at the `SignatureHashSchnorr()` function in
+  `src/script/interpreter.cpp`.
+
+- It's probably most instructive to compare the new `SignatureHashSchnorr()`
+  function to the `SignatureHash()` function immediately below, which
+  calucalates the signature hash for segwit v0 and legacy transactions.
+
+## Questions
+
+1. `SignatureHashSchnorr()` is a templated function. Why?  What types `T` is
+   the template function instantiated with? _Hint: look at how the existing
+   `SignatureHash()` function is called._
+
+2. `SignatureHashSchnorr()` is passed a `PrecomputedTransactionData*` argument.
+   What data is stored in this structure? Why?
+
+3. `SignatureHashSchnorr()` is passed a `hash_type` argument. How many valid
+   hash types are there? The `hash_type` parameter is split into `output_type`
+   and `intput_type` here:
+
+     ```
+     // Hash type
+     const uint8_t output_type = ...
+     const uint8_t input_type = ...
+     ```
+
+     How many valid `output_type`s and `intput_type`s are there?
```

Comment:
- s/intput/input/
- suggest for readability to avoid code formatting in the same word as non-code formatting, e.g. all in `code` or all without

Reply:
- fixed
- reworded the sentence to avoid this.